src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2220 @@
+make sure you have the following matching versions of msgpack and msgpack-numpy installed:
+msgpack==0.5.6, msgpack-numpy==0.4.4
+or
+msgpack==1.0.3 msgpack-numpy==0.4.8
+"""
+
+class FIBSEM(model.HwComponent):
+    """
+    Driver to communicate with autoscript software on TFS microscopes. autoscript is the software TFS uses to control their microscopes.
+    To use this driver the autoscript adapter developed by Delmic should be running on the TFS PC. Communication to the
+    Microscope server is done via Pyro5.
+    """
+
+    def __init__(self, name, role, children, address, port: str ='4242', daemon=None,
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        address: str
+            server ip address for the microscope server (sim address is localhost)
+        port: str
+            server port of the Microscope server, default is '4242'
+        timeout: float
+            Time in seconds the client should wait for a response from the server.
+        """
+
+        model.HwComponent.__init__(self, name, role, daemon=daemon, **kwargs)
+        self._proxy_access = threading.Lock()
+        try:
+            self.server = Pyro5.api.Proxy(f"PYRO:Microscope@{address}:{port}")
+            self.server._pyroTimeout = 30  # seconds
+            self._swVersion = self.server.get_software_version()
+            self._hwVersion = self.server.get_hardware_version()
+            logging.debug(
+                f"Successfully connected to autoscript server with software version {self._swVersion} and hardware"
+                f"version {self._hwVersion}")
+        except CommunicationError as err:
+            raise HwError("Failed to connect to autoscript server '%s'. Check that the "
+                          "uri is correct and autoscript server is"
+                          " connected to the network. %s %s" % (address, err, debug_connection_info))
+        except OSError as err:
+            raise HwError("XT server reported error: %s." % (err,))
+
+        # Transfer latest xtadapter package if available
+        # The transferred package will be a zip file in the form of bytes
+        # self.check_and_transfer_latest_package()
+
+        # Create the scanner type child(ren)
+        # Check if at least one of the required scanner types is instantiated
+        scanner_types = ["sem-scanner", "fib-scanner"]  # All allowed scanners types
+        if not any(scanner_type in children for scanner_type in scanner_types):
+            raise KeyError("FIBSEM was not given any scanner as child. "
+                           "One of 'sem-scanner', 'fib-scanner' need to be included as child")
+
+        has_sem_detector = "sem-detector" in children
+        has_fib_detector = "fib-detector" in children
+
+        if "sem-scanner" in children:
+            kwargs = children["sem-scanner"]
+            self._scanner = Scanner(parent=self, daemon=daemon, channel="electron", has_detector=has_sem_detector, **kwargs)
```

Comment:
```suggestion
            self._scanner = Scanner(parent=self, daemon=daemon, channel="electron", has_detector=True, **kwargs)
```

and line 259 can go?